### PR TITLE
Clean up the project towards more idiomatic Swift and project structure

### PR DESCRIPTION
**Pull request by regexident**
aka CHANGE ALL THE THINGS ™ 😇 

Please note that I haven't yet been able to test/run it locally for a lack of a local GStreamer installation.

### CODE REVIEW

**Comment by aconchillo** [comment]
We should got rid of the `Resources` folder since we don't have the json file anymore.

**Comment by aconchillo** [comment]
Also got these new messages:

Screen Shot 2022-10-13 at 11 17 20 AM

File: cam/FrameSource.swift [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        var bufferPtr = CVPixelBufferGetBaseAddress(pixelBuffer)!
+
+        autoreleasepool {
+            if let data = self.backend.nextFrameBuffer() {
+                let _ = data.withUnsafeBytes { bytes in
+                    memcpy(bufferPtr, bytes.baseAddress, data.count)
+                }
+            }
+
+            if let closure = self.closure {
```
[the remark this answers is not in the document]
Comment (regexident) [reply cam/FrameSource.swift]: Oh, good catch. Looks like I did a last-minute rename of the closure member variable. Fixed.

**Comment by regexident** [comment]
Hm, I don't recall having touched any of the signing bits? 🤔

File: cam/DeviceSource.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+extension DeviceSource: CMIOExtensionDeviceSource {
+    var availableProperties: Set<CMIOExtensionProperty> {
+        return [.deviceTransportType, .deviceModel]
+    }
+
+    func deviceProperties(forProperties properties: Set<CMIOExtensionProperty>) throws -> CMIOExtensionDeviceProperties {
+        let deviceProperties = CMIOExtensionDeviceProperties(dictionary: [:])
+
+        if properties.contains(.deviceTransportType) {
+            deviceProperties.transportType = kIOAudioDeviceTransportTypeVirtual
```
Comment (aconchillo) [inline cam/DeviceSource.swift:26]: ooops, getting:

```
Cannot find 'kIOAudioDeviceTransportTypeVirtual' in scope
```

Comment (regexident) [reply cam/DeviceSource.swift:27]: Looks like I forgot to pull over the `import IOKit.audio`. Fixed.

**Pull request merged**